### ISSUE TITLE
undefined reference to `{elf_getscn, gelf_getshdr, elf_strptr, elf_getdata}'

### ISSUE THREAD

**Issue by scanf** [netoptimizer/prototype-kernel#7]
Hei, 

I wanted to try out the sample programs and seem to be hitting some compile issues. I am probably doing something wrong. Have you seen these errors before?

```
root@builder:~/prototype-kernel/kernel/samples/bpf# make clean
rm -f bpf.o bpf_load.o
rm -f xdp_ddos01_blacklist xdp_ttl xdp_bench01_mem_access_cost
rm -f xdp_ddos01_blacklist_kern.o xdp_ttl_kern.o xdp_bench01_mem_access_cost_kern.o
rm -f xdp_ddos01_blacklist_user.o xdp_ttl_user.o xdp_bench01_mem_access_cost_user.o
root@builder:~/prototype-kernel/kernel/samples/bpf# kbuilddir=/root/linux/ make
gcc -O2 -Wall -I/root/linux//usr/include -I/root/linux//tools/lib -I/root/linux//tools/include -I/root/linux//tools/perf -I/root/linux//tools/testing/selftests/bpf/ -o bpf.o -c /root/linux//tools/lib/bpf/bpf.c
gcc -O2 -Wall -I/root/linux//usr/include -I/root/linux//tools/lib -I/root/linux//tools/include -I/root/linux//tools/perf -I/root/linux//tools/testing/selftests/bpf/ -o bpf_load.o -c /root/linux//samples/bpf/bpf_load.c
gcc -O2 -Wall -I/root/linux//usr/include -I/root/linux//tools/lib -I/root/linux//tools/include -I/root/linux//tools/perf -I/root/linux//tools/testing/selftests/bpf/ -lelf bpf.o bpf_load.o -o xdp_ddos01_blacklist xdp_ddos01_blacklist_user.c
bpf_load.o: In function `get_sec.isra.1':
bpf_load.c:(.text+0x3c): undefined reference to `elf_getscn'
bpf_load.c:(.text+0x4c): undefined reference to `gelf_getshdr'
bpf_load.c:(.text+0x94): undefined reference to `elf_strptr'
bpf_load.c:(.text+0xb4): undefined reference to `elf_getdata'
bpf_load.c:(.text+0xc8): undefined reference to `elf_getdata'
bpf_load.o: In function `load_bpf_file':
bpf_load.c:(.text+0x808): undefined reference to `elf_version'
bpf_load.c:(.text+0x868): undefined reference to `elf_begin'
bpf_load.c:(.text+0x87c): undefined reference to `gelf_getehdr'
bpf_load.c:(.text+0xcbc): undefined reference to `gelf_getrel'
bpf_load.c:(.text+0xcd4): undefined reference to `gelf_getsym'
collect2: error: ld returned 1 exit status
Makefile:152: recipe for target 'xdp_ddos01_blacklist' failed
make: *** [xdp_ddos01_blacklist] Error 1
root@builder:~/prototype-kernel/kernel/samples/bpf# ls ~/
linux  linux-4.10-rc8.tar.xz  prototype-kernel  Scripts
```

**Comment by scanf** [comment]
Please ignore. Tried again on a different machine and did not see those issues.

**Comment by netoptimizer** [comment]
This is likely fixed in pull request #8 by @gospo see commit db6bf73
